### PR TITLE
Spelling fixes and codespell CI job

### PR DESCRIPTION
**Pull request by candrews**
* Add pre-submit PR action prow job for codespell
* chore: Fix spelling errors

Following up on https://github.com/aws-controllers-k8s/iam-controller/pull/78

### CODE REVIEW

File: prow/jobs/jinja/presubmits/service_tests.jinja2 [hunk 1]
```diff
@@ -102,6 +102,27 @@
...
+  - name: {{ service }}-codespell
+    decorate: true
+    optional: false
+    always_run: true
+    annotations:
+      karpenter.sh/do-not-evict: "true"
+    spec:
+      serviceAccountName: pre-submit-service-account
+      containers:
+      - image: docker pull ghcr.io/codespell-project/actions-codespell/stable:v1
```
Comment (RedbackThomson) [inline prow/jobs/jinja/presubmits/service_tests.jinja2:114]: Please update to remove `docker pull`
```suggestion
- image: ghcr.io/codespell-project/actions-codespell/stable:v1
```

**Review by RedbackThomson — CHANGES_REQUESTED** [review]

**Comment by RedbackThomson** [comment]
Thank you for adding the codespell job to the `.jinja` template file. There is one additional step required for this pull request to be complete, and that would be to generate the ProwJobs using the new template. Check out the [`README`](https://github.com/aws-controllers-k8s/test-infra/blob/main/prow/jobs/README.md) for how to do this. If not, I can give you a hand with it.

**Comment by RedbackThomson** [comment]
@candrews We would still love to have these changes. Do you have a chance to make the changes I requested above?

**Comment by candrews** [comment]
> @candrews We would still love to have these changes. Do you have a chance to make the changes I requested above?

Thank you for the ping - this fell of my radar :)

I've made the suggested changes and updated this PR accordingly.

How's it look now?

**Comment by RedbackThomson** [comment]
Oh unfortunately that change messed up the indentation of the jinja job, because Github strips leading whitespace in its suggestions.

Also as well as updating the jinja, we will also need to regenerate all the jobs. See my previous comment for details on that

**Comment by candrews** [comment]
> Oh unfortunately that change messed up the indentation of the jinja job, because Github strips leading whitespace in its suggestions.

I should have noticed that too!

> Also as well as updating the jinja, we will also need to regenerate all the jobs. See my previous comment for details on that

I've made that change and ran `make build-prow-jobs`

File: prow/jobs/jinja/presubmits/service_tests.jinja2 [hunk 2]
```diff
@@ -102,6 +102,27 @@
...
+    always_run: true
+    annotations:
+      karpenter.sh/do-not-evict: "true"
+    spec:
+      serviceAccountName: pre-submit-service-account
+      containers:
+- image: ghcr.io/codespell-project/actions-codespell/stable:v1
+        resources:
+          limits:
+            cpu: 4
```
Comment (A-Hilaly) [inline prow/jobs/jinja/presubmits/service_tests.jinja2:117]: Maybe cpu 4/500Mi is a little too much here?

**Review by A-Hilaly — APPROVED** [review]
Niice! thank you @candrews

File: prow/jobs/jinja/presubmits/service_tests.jinja2 (hunk 2 above)
Comment (candrews) [reply prow/jobs/jinja/presubmits/service_tests.jinja2:117]: Perhaps? Do you have a suggestion?

Comment (RedbackThomson) [reply prow/jobs/jinja/presubmits/service_tests.jinja2:117]: 500Mi should be more than enough. We don't want to waste too many resources on code spelling checks if we don't have to

Comment (candrews) [reply prow/jobs/jinja/presubmits/service_tests.jinja2:117]: I reduced it to 1/200Mi

**Comment by candrews** [comment]
Rebased and resolved merge conflicts

**Comment by RedbackThomson** [comment]
Nice. Looking good!
/lgtm

**Pull request merged**